.circleci/docker/build.sh

```diff
@@ -29,10 +20,42 @@ elif [[ "$image" == *-bionic* ]]; then
   UBUNTU_VERSION=18.04
 elif [[ "$image" == *-focal* ]]; then
   UBUNTU_VERSION=20.04
+elif [[ "$image" == *ubuntu* ]]; then
+  UBUNTU_VERSION="$(echo "${image}" | perl -n -e'/ubuntu(\d+\.\d+)/ && print $1')"
```

Comment:
Can `$(echo "${image}" | perl -n -e'/${pattern}(\d+\.\d+)/ && print $1')` be refactored into a function?

Because we are doing the same version extraction for `ubuntu`, `py`, `cuda`,`cudnn`,`gcc`,`clang`,rocm`, etc...

Reply:
Done.